lib/dry/types/map.rb

```diff
@@ -0,0 +1,104 @@
+module Dry
+  module Types
+    class Map < Definition
+
+      def initialize(keys: Any, values: Any, meta: EMPTY_HASH)
```

Comment:
@flash-gordon @GustavoCaso 
regarding naming:

`Map` is a top level type like `Integer`, `String`, `Array`, `Enum`, ... they are all singular.

But it doesn't seem like you have a strong pattern for plural type-init args.

e.g. array takes the singular `member`, but enum takes the plural `values`.

Personally I am not a huge fan of `keys` and `values`, but I think the singular `key` and `value` is even more confusing.

Do you have any other ideas? Maybe `key_type` and `value_type`?

Reply:
@fledman I also think that `key` and `value` could be confusing. I'm not against having `key_type` and `value_type`.

Reply:
`key_type`/`value_type` is fine